src/main/groovy/com/deploygate/gradle/plugins/tasks/UploadTask.groovy

```diff
@@ -36,6 +44,32 @@ class UploadTask extends DefaultTask {
         onAfterUpload(res)
     }
 
+    private void buildApkFromAppBundleIfNeeded(DeployTarget target) {
+        if (appBundleInfo) {
+            def outputDir = appBundleInfo.getApksFile().parentFile
+
+            outputDir.mkdirs()
+
+            def universalApkGenerateCommand = BuildApksCommand.builder()
+                    .setGenerateOnlyUniversalApk(true)
+                    .setOverwriteOutput(true)
+                    .setBundlePath(appBundleInfo.getBundleFile(project).toPath())
+                    .setSigningConfiguration(appBundleInfo.signingConfig.toSigningConfiguration())
+                    .setOutputFile(appBundleInfo.getApksFile().toPath())
+                    .setAapt2Command(Aapt2Command.createFromExecutablePath(new File(AndroidPlatformUtils.getAapt2Location(project)).toPath()))
+                    .build()
+
+            def path = universalApkGenerateCommand.execute()
+            def p = "unzip -d ${outputDir} ${path}".execute()
```

Comment:
We should aware the path may contain whitespace. Consider using `String[]#execute()`

```groovy
['unzip', '-d', outputDir, path].execute()
```

Reply:
Or better to consider using `java.util.zip.ZipFile` in terms of the portability for Windows. Maybe later.

Reply:
Indeed. let me use `java.util.zip.ZipFile` way.